### PR TITLE
chore: Make inactive posting buttons/fields for "posting disabled" courses

### PR DESCRIPTION
**Pull request by eyatsenkoperpetio**
In some courses (with "Archived" status) posting in Discussions is disabled ("Posting in discussions is disabled by the course team"). On the Web and in edX app all make Post/Response/Comment options are hidden or inactive. We also need to disable (or hide) the "Create new post" button and the "Add a response" / "Add a comment" fields for these courses.

[Make inactive posting buttons/fields for "posting disabled" courses #350](https://github.com/openedx/openedx-app-ios/issues/350)
[Increase tappable area size for sequence elements in Coursework #353](https://github.com/openedx/openedx-app-ios/issues/353)

### CODE REVIEW

**Comment by rnr** [comment]
I think we need disable exactly these options: Post/Response/Comment but not hide. @saeedbashir  what do you think?

**Comment by saeedbashir** [comment]
> I think we need to specifically disable these options: Post/Response/Comment, but not hide. @saeedbashir what do you think?

I've discussed it with @moiz994 and he was okay with hiding the buttons. The only concern was about the banner(`Posting in discussion is disabled by the course team`) design, because we are not using this king of banner in the app. We might need approval from Sam about this, or maybe we need to highlight it so it will be on the design team radar and it will be redesigned post MVP.

File: Discussion/Discussion/Domain/Model/DiscussionInfo.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public var blackouts: [DiscussionBlackout]?
+
+    public func isBlackedOut() -> Bool {
+        guard let blackouts = blackouts else {
+            return false
+        }
+        var isBlackedOut = false
+        for blackout in blackouts {
+            let start = Date(iso8601:  blackout.start)
+            let end = Date(iso8601:  blackout.end)
```
Comment (saeedbashir) [inline Discussion/Discussion/Domain/Model/DiscussionInfo.swift:21]: extra space after `:`

File: Discussion/Discussion/Presentation/Comments/Responses/ResponsesView.swift [hunk 2]
```diff
@@ -249,7 +253,8 @@ struct ResponsesView_Previews: PreviewProvider {
             commentID: "",
             viewModel: viewModel,
             router: router,
-            parentComment: post
+            parentComment: post,
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Comments/Responses/ResponsesView.swift:256]: extra white space st the end

File: Discussion/Discussion/Presentation/Comments/Responses/ResponsesView.swift [hunk 3]
```diff
@@ -259,7 +264,8 @@ struct ResponsesView_Previews: PreviewProvider {
...
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Comments/Responses/ResponsesView.swift:267]: Same https://github.com/openedx/openedx-app-ios/pull/377/files#r1546073019

File: Discussion/Discussion/Presentation/Comments/Responses/ResponsesViewModel.swift [hunk 4]
```diff
@@ -13,9 +13,10 @@ import Combine
 public class ResponsesViewModel: BaseResponsesViewModel, ObservableObject {
     
     @Published var scrollTrigger: Bool = false
-    
     private let threadStateSubject: CurrentValueSubject<ThreadPostState?, Never>
-    
+
+    public var isBlackedOut: Bool = false
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Comments/Responses/ResponsesViewModel.swift:18]: Do we need to have it in viewModel? isn't it good to declare a variable in the view and use it directly

File: Discussion/Discussion/Presentation/DiscussionTopics/DiscussionTopicsViewModel.swift [hunk 5]
```diff
@@ -31,46 +32,54 @@ public class DiscussionTopicsViewModel: ObservableObject {
...
-    
+
     func generateTopics(topics: Topics?) -> [DiscussionTopic] {
         var result = [
             DiscussionTopic(
                 name: DiscussionLocalization.Topics.allPosts,
                 action: {
-                    self.analytics.discussionAllPostsClicked(courseId: self.courseID,
-                                                                    courseName: self.title)
+                    self.analytics.discussionAllPostsClicked(
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/DiscussionTopics/DiscussionTopicsViewModel.swift:55]: As we are here how about using weak self in block?

Comment (eyatsenkoperpetio) [reply Discussion/Discussion/Presentation/DiscussionTopics/DiscussionTopicsViewModel.swift:55]: We don't need weak self for button action from view

File: Discussion/Discussion/Presentation/Comments/Responses/ResponsesViewModel.swift (hunk 4 above)
Comment (eyatsenkoperpetio) [reply Discussion/Discussion/Presentation/Comments/Responses/ResponsesViewModel.swift:17]: I think better use only state variable in view, others variable in view model, and view model say how view should look

**Review by saeedbashir — APPROVED** [review]
LGTM 👍

**Review by volodymyr-chekyrta — APPROVED** [review]
👍

**Review by rnr — APPROVED** [review]

**Pull request merged**